### PR TITLE
Make storage UI close upon being locked

### PR DESCRIPTION
**Pull request by SphiraI**
## About the PR

Made UI's close for people looking at them if the item gets locked.
*Thanks to ShadowCommander for help*

## Why / Balance
Somethin just ain't right about being able to peek into a storage that is actively locked (aka, it's a bug.)

## Technical details
- subscribed `SharedStorageSystem` to `LockToggledEvent` and made a method to close the UI for everyone but admins

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

### CODE REVIEW

File: Content.Shared/Storage/EntitySystems/SharedStorageSystem.cs [hunk 1]
```diff
@@ -1401,6 +1402,26 @@ public ItemSizePrototype GetMaxItemSize(Entity<StorageComponent?> uid)
         return _nextSmallest[item.Size];
     }
 
+    /// <summary>
+    /// Checks if a storage's UI is open  by anyone when locked, and closes it unless they're an admin.
```
Comment (ShadowCommander) [inline Content.Shared/Storage/EntitySystems/SharedStorageSystem.cs:1406]:
```suggestion
    /// Checks if a storage's UI is open by anyone when locked, and closes it unless they're an admin.
```

File: Content.Shared/Storage/EntitySystems/SharedStorageSystem.cs [hunk 2]
```diff
@@ -1401,6 +1402,26 @@ public ItemSizePrototype GetMaxItemSize(Entity<StorageComponent?> uid)
...
+    /// </summary>
+    private void OnLockToggled(EntityUid uid, StorageComponent component, ref LockToggledEvent args)
+    {
+        if (!args.Locked)
+            return;
+
+        //gets everyone looking at the UI
```
Comment (ShadowCommander) [inline Content.Shared/Storage/EntitySystems/SharedStorageSystem.cs:1413]:
```suggestion
        // Gets everyone looking at the UI
```

File: Content.Shared/Storage/EntitySystems/SharedStorageSystem.cs [hunk 3]
```diff
@@ -1401,6 +1402,26 @@ public ItemSizePrototype GetMaxItemSize(Entity<StorageComponent?> uid)
...
+        var actors = _ui.GetActors(uid, StorageComponent.StorageUiKey.Key).ToList();
+        for (var i = actors.Count - 1; i >= 0; i--)
+        {
+            var actor = actors[i];
+            if (_admin.HasAdminFlag(actor, AdminFlags.Admin))
+                continue;
+            //and closes it unless they're an admin
```
Comment (ShadowCommander) [inline Content.Shared/Storage/EntitySystems/SharedStorageSystem.cs:1420]:
```suggestion
            // And closes it unless they're an admin
```

**Review by ShadowCommander — APPROVED** [review]
Just typos in the comments. Otherwise it looks good.

**Review by ShadowCommander — APPROVED** [review]

File: Content.Shared/Storage/EntitySystems/SharedStorageSystem.cs [hunk 4]
```diff
@@ -1401,6 +1402,26 @@ public ItemSizePrototype GetMaxItemSize(Entity<StorageComponent?> uid)
...
+        // Gets everyone looking at the UI
+        var actors = _ui.GetActors(uid, StorageComponent.StorageUiKey.Key).ToList();
+        for (var i = actors.Count - 1; i >= 0; i--)
+        {
+            var actor = actors[i];
+            if (_admin.HasAdminFlag(actor, AdminFlags.Admin))
+                continue;
+            // And closes it unless they're an admin
+            _ui.CloseUi(uid, StorageComponent.StorageUiKey.Key, actor);
+        }
```
Comment (mirrorcult) [inline Content.Shared/Storage/EntitySystems/SharedStorageSystem.cs:1422]:
```suggestion
        // Gets everyone looking at the UI
        foreach (var actor in _ui.GetActors(uid, StorageComponent.StorageUiKey.Key))
        {
            if (_admin.HasAdminFlag(actor, AdminFlags.Admin))
                continue;
                
            // And closes it unless they're an admin
            _ui.CloseUi(uid, StorageComponent.StorageUiKey.Key, actor);
        }
```

just use foreach

**Review by mirrorcult — CHANGES_REQUESTED** [review]

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**